### ISSUE TITLE
Support for Remote Mute Alarms

### ISSUE THREAD

**Issue by zhudhjen** [iMicknl/ha-nest-protect#25]
### Description

Hi Devs, thank you so much for bringing this great integration to everyone! 

So I want to learn about a detailed feature that seems not mentioned in README. Does this integration support remote silencing alarm now? If not, is this on the roadmap that we are going to support? Or is it just not possible for now due to API limitations?

### Device (optional)

Nest Protect

**Comment by iMicknl** [comment]
Is this supported within the Nest App? If so; in theory this could be supported, however don't expect it any time soon. In order to execute commands, we would need to use a different (protobuf) API and I didn't really dig into that.

**Comment by operinko** [comment]
It is supported in the Nest App (as in, you first get a notification as soon as the alarm says "Caution, smoke in X room. You might hear an alarm". And that notification has an option to silence the alarm already. Same effect as pushing the physical button at that point of the alert cycle.

**Comment by mattdm** [comment]
For what it's worth:

1. Our kitchen isn't ventilated well enough, so whenever we burn toast I use this feature.
2. However... I'm concerned about the ramifications of making it easy to silence alarms via automation. Too easy to be clever with "If my smart toaster has been active in the last five minutes, automatically silence any alarms in the kitchen" — and then your house burns down.
3. So, it doesn't seem terrible to me to need to require the Nest app for this specific thing.

**Comment by iMicknl** [comment]
I figured out how to the update endpoint works, so this should be possible. Does someone know if it is also possible to remote mute via the Nest web app? There it is relatively easy to look at the HTTP requests and send me the payload, so I know what to implement.

**Comment by dgriswo** [comment]
You must be within Bluetooth range of the Nest Protect that triggered the alarm to silence it via the app.  There isn't a method to silence via home.nest.com.  https://support.google.com/googlenest/answer/9249356?hl=en  

Documentation isn't clear if it uses Bluetooth to silence the Nest directly, or if it detects the Protect device via Bluetooth and silences via an API call to Nest.

If you are too far away or have Bluetooth disabled, this is the result in the app.
![image](https://user-images.githubusercontent.com/15717486/158043933-83fa74d9-8eab-4a99-89ac-180658f18e82.png)

**Comment by iMicknl** [comment]
Thanks @dgriswo! Eventually, this would require something like Wireshark or Burp suite to investigate or bluetooth is really required. (or if there is an API endpoint that is called after the app validated that you are in Bluetooth reach).

So in short, I don't think this will be integrated very soon, only if someone finds out how this could work.

**Comment by altodd** [comment]
I could easily test this, I already have a proxy set up, I could see if there is an endpoint called when silenced. Is there an easy way to get it to pop up the silence dialog other than burning some toast or something?

**Comment by iMicknl** [comment]
@altodd I don't think so... I didn't trigger mine yet, that's why I didn't know how this remote mute would work :D.

**Comment by blue-hound** [comment]
I have to agree with @mattdm, I've had my fair share of fun debugging automations that stopped working or did fun things that I didn't expect. For example, when my dogs monthly medications didn't get marked as given when I scanned an RFID tag because the integration author changed a variable from a string to an integer. I think silencing a life safety device automatically is dangerous and that ability shouldn't be integrated because it could cost someone their life.

57% of house fire related deaths in the US are attributed to no working or failed smoke detectors. I don't think this integration should tempt fate and risk adding to those numbers. We just had an entire family killed nearby because they didn't have a working smoke detector or it failed to sound. 

This is also one of the reasons that Nest removed the wave to hush feature that the first generation Nest Protects had because real alarms were accidentally silenced and even removed them from the market temporarily (and allowed customers to return them for a full refund if they wanted).

**Comment by dgriswo** [comment]
@blue-hound You make valid points, but I have to oddly disagree with excluding the feature.

FLOSS: Free, **Libre**, Open Source Software.  The user should have the choice to disable a safety feature.  The responsibility as developers would be to convey the risks and dangers of the setting.  If a user intentionally disables the alarm, they are accepting both the risk and liability on their own behalf.  It is akin to removing the batteries from a traditional smoke detector.  Sure, not a smart thing to do, but is certainly possible.

Also, there are certain conditions that Nest will not allow silencing.  "High smoke levels" is one of them: https://support.google.com/googlenest/answer/9249356?hl=en-AU

**Comment by mattdm** [comment]
> FLOSS: Free, Libre, Open Source Software. The user should have the choice to disable a safety feature.

I think this gets into a rather philosophical argument that isn't really best for this github ticket. Happy to take further discussion to Twitter or somewhere.

I will just say: choosing a free software license or otherwise to participate in open source does not obligate a developer to adopt a certain attitude about providing users options _in the software produced_ (either for or against). It means that _you_ have the right to take it and add (or remove!) whatever features and customization you like, which is a very different thing.

**Comment by blue-hound** [comment]
I guess that is up to the developer and what he can live with. I'm a paramedic and a former firefighter. I am a hobby developer, I'd hate to read that something I developed may have contributed to someone's demise. 

That's the last I'll say on the matter, I was just adding to the discussion and maybe pointing out something that the developer hadn't thought of. 

It may also be a reason Google might lock down the Nest API further and the reason it isn't exposed on the Google Developer API we use for the Nest thermostat.

**Comment by iMicknl** [comment]
Thanks all for your input and for sharing your thoughts. I did not look into this issue yet, however if I would know how to remote mute an alarm, I am not against implementing this.

I can imagine that people would like to send an emergency notification (iOS) via Home Assistant with the same functionality as the Nest app offers, thus a remote mute button. It is a shame that Nest didn't implement the emergency push message itself though.

I would never advise to use such a feature in automations since this does not make sense and will indeed make it more dangerous. However, I don't think this will be implemented anytime soon, thus no worries yet.

**Comment by marlarius** [comment]
We just bought a Nest Protect for our hallway.

My use case for this mute feature is having a physical button like an IKEA button to silence the alarm when we fry bacon. I would never make an automation to automatically switch the alarm of.

I still hope this feature will be implemented and I would certainly like to assist with testing etc.

Regarding the bluetooth dependency. What if, after getting the heads up alert with bluetooth enabled, you disable bluetooth and then click the button? If it works, we know that bluetooth is not actually required to mute the device. If it doesn't work, we still can't be sure if the command is sent over bluetooth or it just won't send it if the device is out of bluetooth reach.

**Comment by agambon** [comment]
Just wanted to give a +1 to desiring this feature. My wife is quite short, and when she cooks and the smoke alarm goes off, she can't reach the device to silence it. It would be handy to have an additional button lower down.

**Comment by SolusCado** [comment]
+1 for me too. I  have the exact same issue as @agambon - we already have a Zibgee rotary dial above our stove - it would be FAR more convenient for my wife to be able to silence the smoke alarm with a long press on this dial than to have to get a chair to be able to reach the smoke alarm.

**Comment by jazzyisj** [comment]
@SolusCado in the meantime maybe keep a phone or table with the Nest app on it handy when you're cooking so you're not making your wife climb up onto chairs when she's trying to cook you dinner!  :-)
